app/labor/cache_buster.rb

```diff
@@ -206,3 +247,12 @@ def self.bust_article_comment(commentable)
     end
   end
 end
+
+# Creates our own purge method for an HTTP request,
+# which is used by Nginx to bust a cache.
+# See Net::HTTPGenericRequest for attributes/methods.
+class Net::HTTP::NginxPurge < Net::HTTPRequest # rubocop:disable Style/ClassAndModuleChildren
+  METHOD = "PURGE".freeze
+  REQUEST_HAS_BODY = false
+  RESPONSE_HAS_BODY = true
+end
```

Comment:
We can't use `HTTParty` here the way that we do in `bust_fastly_cache` because `PURGE` is not supported by that client (nor is it defined in the HTTP RFCs, apparently!!).

Took a cue from [`fastly-ruby`](https://github.com/fastly/fastly-ruby), which basically creates their [own version of a `PURGE` Net::HTTPRequest class](https://github.com/fastly/fastly-ruby/blob/159ae26e4fe41bcdef2fc8f45c25d7dd4f0a5cbe/lib/fastly/client.rb#L180-L185). I just named it differently since we know we're only ever going to use it for purging nginx 😉

Reply:
I initially thought one could do extend the HTTParty module with an initializer:

by creating `config/initializers/httparty.rb`:

```ruby
class Net::HTTP::Purge < Net::HTTPRequest # rubocop:disable Style/ClassAndModuleChildren
  METHOD = "PURGE".freeze
  REQUEST_HAS_BODY = false
  RESPONSE_HAS_BODY = true
end

module HTTPartyPurge
  extend ActiveSupport::Concern

  class_methods do
    def purge(path, options = {}, &block)
      perform_request Net::HTTP::Purge, path, options, &block
    end
  end
end

HTTParty.send(:include, HTTPartyPurge)
```

and then you could do:

```ruby
HTTParty.purge("https://dev.to/api")
```

unfortunately the `perform_request` method is private, not sure if there's a way to call it anyway:

https://github.com/jnunemaker/httparty/blob/0d808c7f3a68b8d091506e22003fc00dd84bf6a6/lib/httparty.rb#L556-L561

In the meantime, could we at least put the `Net::HTTP::Purge` class a `net_http.rb` initializer ?

All non blocking obviously :D